{
	"document": {
		"aggregate_severity": {
			"namespace": "https://www.suse.com/support/security/rating/",
			"text": "important"
		},
		"category": "csaf_vex",
		"csaf_version": "2.0",
		"distribution": {
			"text": "Copyright 2025 SUSE LLC. All rights reserved.",
			"tlp": {
				"label": "WHITE",
				"url": "https://www.first.org/tlp/"
			}
		},
		"lang": "en",
		"notes": [
			{
				"category": "summary",
				"text": "SUSE CVE-2019-20478",
				"title": "Title"
			},
			{
				"category": "description",
				"text": "In ruamel.yaml through 0.16.7, the load method allows remote code execution if the application calls this method with an untrusted argument. In other words, this issue affects developers who are unaware of the need to use methods such as safe_load in these use cases.",
				"title": "Description of the CVE"
			},
			{
				"category": "legal_disclaimer",
				"text": "CSAF 2.0 data is provided by SUSE under the Creative Commons License 4.0 with Attribution (CC-BY-4.0).",
				"title": "Terms of use"
			}
		],
		"publisher": {
			"category": "vendor",
			"contact_details": "https://www.suse.com/support/security/contact/",
			"name": "SUSE Product Security Team",
			"namespace": "https://www.suse.com/"
		},
		"references": [
			{
				"category": "external",
				"summary": "CVE-2019-20478",
				"url": "https://www.suse.com/security/cve/CVE-2019-20478"
			},
			{
				"category": "external",
				"summary": "SUSE Security Ratings",
				"url": "https://www.suse.com/support/security/rating/"
			},
			{
				"category": "external",
				"summary": "SUSE Bug 1164458 for CVE-2019-20478",
				"url": "https://bugzilla.suse.com/1164458"
			}
		],
		"title": "SUSE CVE CVE-2019-20478",
		"tracking": {
			"current_release_date": "2025-01-01T06:06:05Z",
			"generator": {
				"date": "2023-02-15T04:05:18Z",
				"engine": {
					"name": "cve-database.git:bin/generate-csaf-vex.pl",
					"version": "1"
				}
			},
			"id": "CVE-2019-20478",
			"initial_release_date": "2023-02-15T04:05:18Z",
			"revision_history": [
				{
					"date": "2023-02-15T04:05:18Z",
					"number": "2",
					"summary": "Current version"
				},
				{
					"date": "2024-07-13T03:44:42Z",
					"number": "3",
					"summary": "Current version"
				},
				{
					"date": "2025-01-01T06:06:05Z",
					"number": "4",
					"summary": "Current version"
				}
			],
			"status": "interim",
			"version": "4"
		}
	},
	"product_tree": {
		"branches": [
			{
				"branches": [
					{
						"branches": [
							{
								"category": "product_name",
								"name": "openSUSE Tumbleweed",
								"product": {
									"name": "openSUSE Tumbleweed",
									"product_id": "openSUSE Tumbleweed",
									"product_identification_helper": {
										"cpe": "cpe:/o:opensuse:tumbleweed"
									}
								}
							},
							{
								"category": "product_version",
								"name": "python310-ruamel.yaml-0.18.6-1.2",
								"product": {
									"name": "python310-ruamel.yaml-0.18.6-1.2",
									"product_id": "python310-ruamel.yaml-0.18.6-1.2"
								}
							},
							{
								"category": "product_version",
								"name": "python311-ruamel.yaml-0.18.6-1.2",
								"product": {
									"name": "python311-ruamel.yaml-0.18.6-1.2",
									"product_id": "python311-ruamel.yaml-0.18.6-1.2"
								}
							},
							{
								"category": "product_version",
								"name": "python312-ruamel.yaml-0.18.6-1.2",
								"product": {
									"name": "python312-ruamel.yaml-0.18.6-1.2",
									"product_id": "python312-ruamel.yaml-0.18.6-1.2"
								}
							},
							{
								"category": "product_version",
								"name": "python36-ruamel.yaml-0.17.10-1.2",
								"product": {
									"name": "python36-ruamel.yaml-0.17.10-1.2",
									"product_id": "python36-ruamel.yaml-0.17.10-1.2",
									"product_identification_helper": {
										"cpe": "cpe:2.3:a:ruamel.yaml_project:ruamel.yaml:0.17.10:*:*:*:*:*:*:*"
									}
								}
							},
							{
								"category": "product_version",
								"name": "python38-ruamel.yaml-0.17.10-1.2",
								"product": {
									"name": "python38-ruamel.yaml-0.17.10-1.2",
									"product_id": "python38-ruamel.yaml-0.17.10-1.2",
									"product_identification_helper": {
										"cpe": "cpe:2.3:a:ruamel.yaml_project:ruamel.yaml:0.17.10:*:*:*:*:*:*:*"
									}
								}
							},
							{
								"category": "product_version",
								"name": "python39-ruamel.yaml-0.17.10-1.2",
								"product": {
									"name": "python39-ruamel.yaml-0.17.10-1.2",
									"product_id": "python39-ruamel.yaml-0.17.10-1.2",
									"product_identification_helper": {
										"cpe": "cpe:2.3:a:ruamel.yaml_project:ruamel.yaml:0.17.10:*:*:*:*:*:*:*"
									}
								}
							}
						],
						"category": "product_family",
						"name": "SUSE Linux Enterprise"
					}
				],
				"category": "vendor",
				"name": "SUSE"
			}
		],
		"relationships": [
			{
				"category": "default_component_of",
				"full_product_name": {
					"name": "python310-ruamel.yaml-0.18.6-1.2 as component of openSUSE Tumbleweed",
					"product_id": "openSUSE Tumbleweed:python310-ruamel.yaml-0.18.6-1.2"
				},
				"product_reference": "python310-ruamel.yaml-0.18.6-1.2",
				"relates_to_product_reference": "openSUSE Tumbleweed"
			},
			{
				"category": "default_component_of",
				"full_product_name": {
					"name": "python311-ruamel.yaml-0.18.6-1.2 as component of openSUSE Tumbleweed",
					"product_id": "openSUSE Tumbleweed:python311-ruamel.yaml-0.18.6-1.2"
				},
				"product_reference": "python311-ruamel.yaml-0.18.6-1.2",
				"relates_to_product_reference": "openSUSE Tumbleweed"
			},
			{
				"category": "default_component_of",
				"full_product_name": {
					"name": "python312-ruamel.yaml-0.18.6-1.2 as component of openSUSE Tumbleweed",
					"product_id": "openSUSE Tumbleweed:python312-ruamel.yaml-0.18.6-1.2"
				},
				"product_reference": "python312-ruamel.yaml-0.18.6-1.2",
				"relates_to_product_reference": "openSUSE Tumbleweed"
			},
			{
				"category": "default_component_of",
				"full_product_name": {
					"name": "python36-ruamel.yaml-0.17.10-1.2 as component of openSUSE Tumbleweed",
					"product_id": "openSUSE Tumbleweed:python36-ruamel.yaml-0.17.10-1.2"
				},
				"product_reference": "python36-ruamel.yaml-0.17.10-1.2",
				"relates_to_product_reference": "openSUSE Tumbleweed"
			},
			{
				"category": "default_component_of",
				"full_product_name": {
					"name": "python38-ruamel.yaml-0.17.10-1.2 as component of openSUSE Tumbleweed",
					"product_id": "openSUSE Tumbleweed:python38-ruamel.yaml-0.17.10-1.2"
				},
				"product_reference": "python38-ruamel.yaml-0.17.10-1.2",
				"relates_to_product_reference": "openSUSE Tumbleweed"
			},
			{
				"category": "default_component_of",
				"full_product_name": {
					"name": "python39-ruamel.yaml-0.17.10-1.2 as component of openSUSE Tumbleweed",
					"product_id": "openSUSE Tumbleweed:python39-ruamel.yaml-0.17.10-1.2"
				},
				"product_reference": "python39-ruamel.yaml-0.17.10-1.2",
				"relates_to_product_reference": "openSUSE Tumbleweed"
			}
		]
	},
	"vulnerabilities": [
		{
			"cve": "CVE-2019-20478",
			"ids": [
				{
					"system_name": "SUSE CVE Page",
					"text": "https://www.suse.com/security/cve/CVE-2019-20478"
				}
			],
			"notes": [
				{
					"category": "general",
					"text": "In ruamel.yaml through 0.16.7, the load method allows remote code execution if the application calls this method with an untrusted argument. In other words, this issue affects developers who are unaware of the need to use methods such as safe_load in these use cases.",
					"title": "CVE description"
				}
			],
			"product_status": {
				"recommended": [
					"openSUSE Tumbleweed:python310-ruamel.yaml-0.18.6-1.2",
					"openSUSE Tumbleweed:python311-ruamel.yaml-0.18.6-1.2",
					"openSUSE Tumbleweed:python312-ruamel.yaml-0.18.6-1.2",
					"openSUSE Tumbleweed:python36-ruamel.yaml-0.17.10-1.2",
					"openSUSE Tumbleweed:python38-ruamel.yaml-0.17.10-1.2",
					"openSUSE Tumbleweed:python39-ruamel.yaml-0.17.10-1.2"
				]
			},
			"references": [
				{
					"category": "external",
					"summary": "CVE-2019-20478",
					"url": "https://www.suse.com/security/cve/CVE-2019-20478"
				},
				{
					"category": "external",
					"summary": "SUSE Security Ratings",
					"url": "https://www.suse.com/support/security/rating/"
				},
				{
					"category": "external",
					"summary": "SUSE Bug 1164458 for CVE-2019-20478",
					"url": "https://bugzilla.suse.com/1164458"
				}
			],
			"remediations": [
				{
					"category": "vendor_fix",
					"details": "To install this SUSE Security Update use the SUSE recommended installation methods like YaST online_update or \"zypper patch\".\n",
					"product_ids": [
						"openSUSE Tumbleweed:python310-ruamel.yaml-0.18.6-1.2",
						"openSUSE Tumbleweed:python311-ruamel.yaml-0.18.6-1.2",
						"openSUSE Tumbleweed:python312-ruamel.yaml-0.18.6-1.2",
						"openSUSE Tumbleweed:python36-ruamel.yaml-0.17.10-1.2",
						"openSUSE Tumbleweed:python38-ruamel.yaml-0.17.10-1.2",
						"openSUSE Tumbleweed:python39-ruamel.yaml-0.17.10-1.2"
					]
				}
			],
			"scores": [
				{
					"cvss_v3": {
						"baseScore": 7.3,
						"baseSeverity": "HIGH",
						"vectorString": "CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:L/I:L/A:L",
						"version": "3.1"
					},
					"products": [
						"openSUSE Tumbleweed:python310-ruamel.yaml-0.18.6-1.2",
						"openSUSE Tumbleweed:python311-ruamel.yaml-0.18.6-1.2",
						"openSUSE Tumbleweed:python312-ruamel.yaml-0.18.6-1.2",
						"openSUSE Tumbleweed:python36-ruamel.yaml-0.17.10-1.2",
						"openSUSE Tumbleweed:python38-ruamel.yaml-0.17.10-1.2",
						"openSUSE Tumbleweed:python39-ruamel.yaml-0.17.10-1.2"
					]
				}
			],
			"threats": [
				{
					"category": "impact",
					"date": "2020-02-19T14:45:42Z",
					"details": "important"
				}
			],
			"title": "CVE-2019-20478"
		}
	]
}
